web/pgadmin/static/js/helpers/Menu.js

```diff
@@ -184,9 +159,17 @@ export class MenuItem {
     };
   }
 
-  setDisabled(disabled) {
-    this.is_disabled = disabled;
-    this.onDisableChange?.(this.parentMenu, this);
+  checkAndSetDisabled(node, item, forceDisable) {
+    if(!_.isUndefined(forceDisable)) {
+      this._isDisabled = forceDisable;
+      this.onDisableChange?.(this.parentMenu, this);
+      return;
+    }
+    this._isDisabled = this.disabled(node, item);
```

Comment:
`onDisableChange`  should get called even `forceDisable` is `undefined`, to update the native menus.